proposal/src/main/scala/explore/proposal/PartnerSplitsEditor.scala

```diff
@@ -0,0 +1,135 @@
+// Copyright (c) 2016-2020 Association of Universities for Research in Astronomy, Inc. (AURA)
+// For license information see LICENSE or https://opensource.org/licenses/BSD-3-Clause
+
+package explore.proposal
+
+import cats.syntax.all._
+import eu.timepit.refined.auto._
+import explore.components.ui.ExploreStyles
+import explore.components.ui.PartnerFlags
+import explore.model._
+import explore.model.reusability._
+import japgolly.scalajs.react._
+import japgolly.scalajs.react.vdom.html_<^._
+import japgolly.scalajs.react.vdom.VdomNode
+import lucuma.core.model.Partner
+import mouse.all._
+import react.common.ReactProps
+import react.semanticui.collections.form.Form
+import react.semanticui.elements.button.Button
+import react.semanticui.modules.modal._
+
+final case class PartnerSplitsEditor(
+  show:         Boolean,
+  splits:       List[PartnerSplit],
+  closeMe:      Callback,
+  updateSplits: List[PartnerSplit] => Callback,
+  onSave:       List[PartnerSplit] => Callback
+) extends ReactProps[PartnerSplitsEditor](PartnerSplitsEditor.component)
+
+object PartnerSplitsEditor {
+  type Props = PartnerSplitsEditor
+
+  implicit val propsReuse: Reusability[Props] = Reusability.by(p => (p.show, p.splits))
+
+  private def toolbar(p: Props): ModalActions =
+    ModalActions(
+      <.div(
+        ExploreStyles.FlexContainer,
+        <.div("Total must be 100.",
+              ExploreStyles.PartnerSplitsEditorError,
+              ExploreStyles.FlexGrow(1)
+        )
+          .unless(addsUpTo100(p)),
+        Button(onClick = p.closeMe)(^.tpe := "button")("Cancel", ExploreStyles.FlexEnd),
+        Button(^.tpe := "submit")("OK", ^.disabled := !addsUpTo100(p))
+      )
+    )
+
+  private def updatePercent(p: Props, partner: Partner)(e: ReactEventFromInput): Callback = {
+    val newValue =
+      if (e.target.value === "") "0"
+      else e.target.value
+    newValue.parseIntOption
+      .flatMap(_ match {
+        case i if i >= 0 && i <= 100 => Some(i)
+        case _                       => None
+      })
+      .map { pct =>
+        p.splits.map { s =>
+          if (s.partner === partner) s.copy(percent = pct)
+          else s
+        }
+      }
+      .fold(Callback.empty)(p.updateSplits)
+  }
+
+  private def makeTableRows(p: Props): TagMod =
+    p.splits.zipWithIndex.toTagMod { case (ps, idx) =>
+      val id = s"${ps.partner.tag}-percent"
+      React.Fragment(
+        <.tr(
```

Comment:
Why are we using tables for layout that rather than e.g. grid layout?

Reply:
Because this seems like a semantically appropriate use for the `<table>` element. There is a header, rows of semantically identical data, and a footer. 

The problem with `table` was that people were using it in non-semantic ways for layout because things like CSS flex and grid did not exist. But, this seems like a proper use to me.

Reply:
Sorry I'm commenting so late. It's definitely a table. Could we use the SUI `Table` though? I'd like to SUI widgets whenever possible to ensure consistent CSS is applied throughout, and also to keep a uniform coding style.

Reply:
semantic-ui-react has a component for the SUI table, but there is no facade for it in scalajs-reacte-semantic-ui. Since it is just adding classes, do we need to create a facade for it? I don't think there is any JS functionality in the Table.

Reply:
Yes, we should add facades for `Table` and related components. I had not realized we had't done it yet, but we should.

Even if there's no behavior added, using the SUIR components allows us to define the UI at the semantic level instead of at the HTML level. For example, we can use `basic`, `color`, `stackable`, `striped`, etc, and the behavior of these properties will be coherent with the overall theme for the app.

Reply:
Yeah, I came to the same conclusion myself. I just added the classes to the table/row elements to test it out. I'll probably push that in a bit. Then, I'll add the facade and change it when it's available. I think I'll switch to working on the masked input stuff for Carlos first, though.

Reply:
Sounds good :+1: